File: Javascript/Greedy/M-334. Increasing Triplet Subsequence.ts
function increasingTriplet(nums: number[]): boolean {
  let firstNum = Infinity;
  let secondNum = Infinity;

  for (const currNum of nums) {
    if (currNum > secondNum && secondNum > firstNum) return true;

    //if (currNum > firstNum && currNum <= secondNum) secondNum = currNum;
    if (currNum > firstNum) secondNum = currNum; // equal to line 8
    else firstNum = currNum;
  }

  return false;
}

const testData = [
  { nums: [1, 2, 3, 4, 5], ans: true },
  { nums: [5, 4, 3, 2, 1], ans: false },
  { nums: [2, 1, 5, 2, 3, 6], ans: true },
  { nums: [5, 1, 5, 5, 2, 5, 4], ans: true },
];

for (const data of testData) {
  console.log(data['ans'] === increasingTriplet(data['nums']));
}

File: Greedy/M-334. Increasing Triplet Subsequence/M-334. Increasing Triplet Subsequence.py
import sys


class Solution:
    def increasingTriplet(self, nums: List[int]) -> bool:
        first_num = sys.maxsize
        second_num = sys.maxsize

        for num in nums:
            if num > second_num and second_num > first_num:
                return True

            if num > first_num:
                second_num = num
            else:
                first_num = num

        return False
